src/vcpkg/registries.cpp

```diff
@@ -491,101 +532,95 @@ namespace
     // { GitRegistry::RegistryImplementation
     std::unique_ptr<RegistryEntry> GitRegistry::get_port_entry(const VcpkgPaths& paths, StringView port_name) const
     {
-        auto port_versions = get_versions_tree_path(paths);
-        auto maybe_version_entries =
-            load_versions_file(paths.get_filesystem(), VersionDbType::Git, port_versions, port_name);
-        Checks::check_maybe_upgrade(
-            VCPKG_LINE_INFO, maybe_version_entries.has_value(), "Error: " + maybe_version_entries.error());
-        auto version_entries = std::move(maybe_version_entries).value_or_exit(VCPKG_LINE_INFO);
+        if (!m_paths) m_paths = &paths;
+        return std::make_unique<GitRegistryEntry>(*this, port_name);
+    }
 
-        auto res = std::make_unique<GitRegistryEntry>();
-        res->port_name = port_name.to_string();
-        for (auto&& version_entry : version_entries)
-        {
-            res->port_versions.push_back(version_entry.version);
-            res->git_trees.push_back(version_entry.git_tree);
-        }
-        return res;
+    GitRegistryEntry::GitRegistryEntry(const GitRegistry& reg, StringView name)
+        : port_name(name.to_string()), parent(reg)
+    {
+        // This guarantees that parent.m_paths has been filled before any entries are constructed.
+        // The entries may then rely on (bool)parent.m_paths during their operation
+        Checks::check_exit(VCPKG_LINE_INFO, parent.m_paths);
+#if defined(_MSC_VER)
+        __assume(parent.m_paths);
+#endif
```

Comment:
why is this here. This is not performance sensitive code.

Reply:
Otherwise it fails `/analyze` because the analyzer doesn't know that `Checks::check_exit()` guarantees the condition is true on return.